name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java

```diff
@@ -172,206 +174,134 @@ private void addBuySellTransaction()
-                                        BigDecimal.valueOf(taxRefundFX.getAmount()).multiply(exchangeRate)
-                                            .setScale(0, RoundingMode.HALF_UP).longValue());
-
-                        v.put("taxRefund", BigDecimal.valueOf(taxRefundFX.getAmount(), 2).toString().replace('.', ','));
-
-                        t.setAmount(t.getPortfolioTransaction().getAmount() - asAmount(v.get("taxRefund")));
-                    }
-                })
+                .assign(this::extractTaxRefundDifferentCurrency)
 
                 //                                     ***Einbeh. SichSt EUR                -1,00
                 .section("taxRefund", "currency").optional()
                 .match("^.* \\*\\*\\*Einbeh\\. SichSt ([\\s]+)?(?<currency>[\\w]{3}) ([\\s]+)?-(?<taxRefund>[.,\\d]+)$")
-                .assign((t, v) -> {
-                    if (t.getPortfolioTransaction().getCurrencyCode().equals(v.get("currency")))
-                    {
-                        t.setAmount(t.getPortfolioTransaction().getAmount() - asAmount(v.get("taxRefund")));
-                    }
-                })
+                .assign(this::extractTaxRefundSameCurrency)
 
                 // Devisenkurs   : 1,192200(x)             Provision     :
                 //                                     ***Einbeh. SichSt EUR                -1,00
                 .section("taxRefund", "currency", "exchangeRate").optional()
                 .match("^Devisenkurs ([:\\s]+)?(?<exchangeRate>[.,\\d]+).*$")
                 .match("^.* \\*\\*\\*Einbeh\\. SichSt ([\\s]+)?(?<currency>[\\w]{3}) ([\\s]+)?-(?<taxRefund>[.,\\d]+)$")
+                .assign(this::extractTaxRefundDifferentCurrency)
+
+                .wrap(BuySellEntryItem::new);
+
+        addTaxesSectionsTransaction(pdfTransaction, type);
+        addFeesSectionsTransaction(pdfTransaction, type);
+        addTaxReturnBlock(type);
+
+        pdfTransaction
+                // Storno Wertpapierabrechnung Kauf Fonds/Zertifikate
+                // Stornierung Wertpapierabrechnung Kauf Fonds
+                .section().optional()
+                .match("^(Storno|Stornierung) Wertpapierabrechnung.*$")
                 .assign((t, v) -> {
-                    if (!t.getPortfolioTransaction().getCurrencyCode().equals(v.get("currency")))
-                    {
-                        Money taxRefundFX = Money.of(asCurrencyCode(v.get("currency")), asAmount(v.get("taxRefund")));
+                    invertFeesForRefund((BuySellEntry) t);
+                });
+    }
 
-                        BigDecimal exchangeRate = asExchangeRate(v.get("exchangeRate"));
-                        exchangeRate = BigDecimal.ONE.divide(exchangeRate, 10, RoundingMode.HALF_DOWN);
+    private void invertFeesForRefund(BuySellEntry t) {
+        if (t.getPortfolioTransaction().getType() == PortfolioTransaction.Type.SELL) {
```

Comment:
I can't see the build logs at the moment, but I expect check style will complain about the curly braces to be on the new line

Reply:
Installed checkstyle and fixed the violations.